src/formatDate.js

```diff
@@ -8,7 +8,88 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let result = '';
+  const oldDate = date;
+  const newFormat = toFormat;
```

Comment:
Directly modifying `newFormat` can lead to unexpected behavior if the same format is reused elsewhere. Consider working with a copy of `newFormat` to avoid side effects.